.github/workflows/benchmark-local.yaml

```diff
@@ -0,0 +1,73 @@
+# cli command to run this from repo
+# given a run, get output (link?)
+# input variable to CLI to allow users to specify where to output their artefacts
+
+name: Run performance comparisons
+
+on:
+  workflow_dispatch:
+    inputs:
+      c1:
+        description: First commit hash
+        required: true
+      c2:
+        description: Second commit hash
+        required: true
+
+jobs:
+  build:
+    runs-on: [self-hosted, linux, x64, bench]
+    strategy:
+      fail-fast: true
+      matrix:
+        commit: ["${{ github.event.inputs.c1 }}", "${{ github.event.inputs.c2 }}"]
+    timeout-minutes: 15
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - name: Assume AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-west-2
+        role-to-assume: ${{ secrets.ACTIONS_AWS_ROLE_ARN }}
+        role-session-name: daft-performance-comparisons
+
+    - name: "Checkout commit ${{ matrix.commit }}"
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ matrix.commit }}
+
+    - name: Install uv and python
+      run: |
+        curl -LsSf https://astral.sh/uv/install.sh | sh
+        source $HOME/.cargo/env
+        uv python install 3.9
+        uv python pin 3.9
```

Comment:
We might want to globally pin Python, so that we can ensure consistency later on too with our Ray cluster

Reply:
Hm. Since `uv python pin $PYTHON_VERSION` is not being run in a `.venv`, should this not apply globally?

Reply:
Sorry, I think I meant having this as like a workflow-level input so that we can ensure that the ray cluster launching step also uses the same (exact version, down to the patch version) of Python

Reply:
Oh yes, that has been added in a previous commit. There is an input called `PYTHON_VERSION` which I recently added which addresses this request.